[BUG]formData[2].value: expected string, got number

<!-- ⚠️⚠️ Do not delete this template ⚠️⚠️ -->

<!-- 🔎 Search existing issues to avoid creating duplicates. -->
<!-- 🧪 Test using the latest Playwright release to see if your issue has already been fixed -->
<!-- 💡 Provide enough information for us to be able to reproduce your issue locally -->

### System info
- Playwright Version: [v1.35.0]
- Operating System: [Windows 11]
- Browser: [Chromium]
- Other info:

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

<!-- For simple cases, please provide a self-contained test file along with the config file -->
<!-- For larger cases, you can provide a GitHub repo you created for this issue -->
<!-- If we can not reproduce the problem locally, we won't be able to act on it -->
<!-- You can still file without the exact code and we will try to help, but if we can't repro, it will be closed -->

**Link to the GitHub repository with the repro**

[https://github.com/your_profile/playwright_issue_title]

or

**Test file (self-contained)**

```java
package com.wust.linghang.yp_crawler;

import com.microsoft.playwright.Browser;
import com.microsoft.playwright.BrowserContext;
import com.microsoft.playwright.Playwright;
import com.microsoft.playwright.options.FormData;
import com.microsoft.playwright.options.RequestOptions;

public class Example {
    public static void main(String[] args) {
        Playwright playwright = Playwright.create();
        Browser launch = playwright.firefox().launch();
        BrowserContext browserContext = launch.newContext();
        try (playwright;launch;browserContext){
            FormData form = FormData.create()
                    .set("firstName", "John")
                    .set("lastName", "Doe")
                    .set("age", 30);
            browserContext.newPage().request().post("http://localhost:8080/submit", RequestOptions.create().setForm(form));
        }

    }
}

```

**Steps**
- [Run the test]
- [...]

**Expected**

I want to construct a formData request  with value as a number. such as "age": 30

**Actual**

Caused by: com.microsoft.playwright.impl.DriverException: Error {
  message='formData[2].value: expected string, got number
  name='Error
  stack='Error: formData[2].value: expected string, got number
at tString (C:\Users\97834\AppData\Local\Temp\playwright-java-16320733193796098865\package\lib\protocol\validatorPrimitives.js:48:9)
    at C:\Usersl\Temp\playwright-java-16320733193796098865\package\lib\protocol\validatorPrimitives.js:99:21
    at C:\Users\Temp\playwright-java-16320733193796098865\package\lib\protocol\validator.js:39:14
    at C:\Users\Temp\playwright-java-16320733193796098865\package\lib\protocol\validatorPrimitives.js:86:34
    at Array.map (<anonymous>)

--- a/playwright/src/main/java/com/microsoft/playwright/impl/Serialization.java
+++ b/playwright/src/main/java/com/microsoft/playwright/impl/Serialization.java
@@ -330,6 +330,11 @@ class Serialization {
   }
 
   static JsonArray toProtocol(Map<String, String> map) {
+    for (String value : map.values()) {
+      if (value == null) {
+        throw new PlaywrightException("Value cannot be null");
+      }
+    }
     return toNameValueArray(map);
   }
 

--- a/playwright/src/main/java/com/microsoft/playwright/impl/Serialization.java
+++ b/playwright/src/main/java/com/microsoft/playwright/impl/Serialization.java
@@ -348,7 +348,11 @@ class Serialization {
     for (Map.Entry<String, ?> e : map.entrySet()) {
       JsonObject item = new JsonObject();
       item.addProperty("name", e.getKey());
-      item.add("value", gson().toJsonTree(e.getValue()));
+      if (e.getValue() instanceof FilePayload) {
+        item.add("value", gson().toJsonTree(e.getValue()));
+      } else {
+        item.addProperty("value", "" + e.getValue());
+      }
       array.add(item);
     }
     return array;

--- a/playwright/src/test/java/com/microsoft/playwright/TestBrowserContextFetch.java
+++ b/playwright/src/test/java/com/microsoft/playwright/TestBrowserContextFetch.java
@@ -511,14 +511,18 @@ public class TestBrowserContextFetch extends TestBase {
       FormData.create()
         .set("firstName", "John")
         .set("lastName", "Doe")
+        .set("age", 30)
+        .set("isMale", true)
         .set("file", "f.js")));
 
     assertEquals("POST", req.get().method);
     assertEquals(asList("application/x-www-form-urlencoded"), req.get().headers.get("content-type"));
     String body = new String(req.get().postBody);
-    assertTrue(body.contains("firstName=John"));
-    assertTrue(body.contains("lastName=Doe"));
-    assertTrue(body.contains("file=f.js"));
+    assertTrue(body.contains("firstName=John"), body);
+    assertTrue(body.contains("lastName=Doe"), body);
+    assertTrue(body.contains("age=30"), body);
+    assertTrue(body.contains("isMale=true"), body);
+    assertTrue(body.contains("file=f.js"), body);
   }
 
   @Test
@@ -526,6 +530,8 @@ public class TestBrowserContextFetch extends TestBase {
     Map<String, Object> data = mapOf(
       "firstName", "John",
       "lastName", "Doe",
+      "age", 30,
+      "isMale", true,
       "file", mapOf("name", "f.js")
     );
     Future<Server.Request> req = server.futureRequest("/empty.html");

--- a/playwright/src/test/java/com/microsoft/playwright/TestPageSetExtraHttpHeaders.java
+++ b/playwright/src/test/java/com/microsoft/playwright/TestPageSetExtraHttpHeaders.java
@@ -73,6 +73,6 @@ public class TestPageSetExtraHttpHeaders extends TestBase {
     PlaywrightException e = assertThrows(PlaywrightException.class, () -> {
       browser.newContext(new Browser.NewContextOptions().setExtraHTTPHeaders(mapOf("foo", null)));
     });
-    assertTrue(e.getMessage().contains("expected string, got undefined"));
+    assertTrue(e.getMessage().contains("Value cannot be null"));
   }
 }

